client/src/components/gallery_detail/GalleryDetail.tsx

```diff
@@ -0,0 +1,257 @@
+import React, {
+  FunctionComponent,
+  useState,
+  useMemo,
+} from 'react';
+import { withTheme } from '@material-ui/core/styles';
+import styled from 'styled-components';
+import { useLocation } from 'react-router-dom';
+import Box from '@material-ui/core/Box';
+import Slick, { Settings } from 'react-slick';
+import Paper from '@material-ui/core/Paper';
+import Typography from '@material-ui/core/Typography';
+import map from 'lodash/map';
+import { scroller } from 'react-scroll';
+
+import { Gallery, GalleryImage } from '../../types/gallery';
+import { AutoSlick } from '../slider/AutoSlick';
+
+interface GalleryProps {
+  className?: string;
+}
+
+/**
+ * To make Selected Gallery Slide show
+ */
+const UnstyledGalleryDetail: FunctionComponent<GalleryProps> = ({
+  className,
+}) => {
+  const location = useLocation();
+  const data = location.state as { gallery: Gallery };
+  const gallery: Gallery = data.gallery;
+
+  const [selectedImg, SetSelectedImg] = useState<GalleryImage>();
+
+  const NavSlderSetting = useMemo<Settings>(
+    () => ({
+      dots: false,
+      arrows: false,
+      centerMode: true,
+      slidesToShow: 6,
+      swipeToSlide: true,
+      focusOnSelect: true,
+      autoplay: true,
+    }),
+    []
+  );
+
+  if (!gallery) {
+    return <div>loading...</div>;
+  }
+
+  return (
+    <Box className={className} mt={20}>
+      <Box>
+        <AutoSlick speed={200} arrows={false}>
+          {map(gallery.galleryImages, (img) => (
+            <Box className="slider-item" key={img.id}>
+              <img src={img.imageURL} alt={`${gallery.title}_${img.id}`} />
+            </Box>
+          ))}
+        </AutoSlick>
+
+        {gallery.galleryImages && gallery.galleryImages?.length > 1 && (
+          <Box className="thumbnail-gallery">
+            {gallery.galleryImages.length > 7 ? (
+              <Slick
+                {...NavSlderSetting}
+                className="thumbnail-slider-large"
+                autoplay={false}
+                slidesToShow={Math.min(gallery.galleryImages.length, 6)}
+                arrows={true}
+              >
+                {map(gallery.galleryImages, (img) => (
+                  <Box className="thumbnail-item" key={img.id}>
+                    <img
+                      src={img.imageURL}
+                      alt={img.id}
+                      onClick={() => {
+                        SetSelectedImg(img);
+                        scroller.scrollTo('zoom-box', {
+                          smooth: true,
+                          offset: 0,
+                          duration: 500,
+                          delay: 50,
+                        });
+                      }}
+                    />
+                  </Box>
+                ))}
+              </Slick>
+            ) : (
+              <Box className="thumbnail-slider-smaill">
```

Comment:
please add comments on what all of these checks are responsible for. I have no idea what's going on